oolab/src/main/java/agh/ics/oop/OptionsParser.java

```diff
@@ -0,0 +1,26 @@
+package agh.ics.oop;
+
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class OptionsParser {
+
+    private MoveDirection fromStr(String arg) {
+        return switch(arg) {
+            case "f" -> MoveDirection.FORWARD;
+            case "b" -> MoveDirection.BACKWARD;
+            case "r" -> MoveDirection.RIGHT;
+            case "l" -> MoveDirection.LEFT;
+            default -> null;
+        };
+    }
+
+    public MoveDirection[] parse(String[] arguments) {
+        return Arrays.stream(arguments)
+                .filter(Objects::nonNull)
+                .filter(s -> s.matches("^[fblr]$"))
```

Comment:
to raczej nadmiarowe, mamy już switcha, który zwróci nulla w takich przypadkach -> trzeba by tylko ten filter dla nulli prznieść już pod `.map(this::fromStr)`